eth/ethtest/common_test.go

```diff
@@ -0,0 +1,231 @@
+		sim:           sim,
+		boundContract: boundContract,
+		blockNum:      blockNum,
+		nodeStorage:   nodeStorage,
+		doInOneBlock:  doInOneBlock,
+	}
+}
+
+type TestEnv struct {
+	eventSyncer    *eventsyncer.EventSyncer
+	validators     []*testValidatorData
+	ops            []*testOperator
+	nodeStorage    storage.Storage
+	sim            *simulator.SimulatedBackend
+	boundContract  *simcontract.Simcontract
+	auth           *bind.TransactOpts
+	shares         [][]byte
+	execClient     *executionclient.ExecutionClient
+	rpcServer      *rpc.Server
+	httpSrv        *httptest.Server
+	validatorCtrl  *mocks.MockController
+	mockCtrl       *gomock.Controller
+	followDistance *uint64
+}
+
+func (e *TestEnv) shutdown() {
+	if e.mockCtrl != nil {
+		e.mockCtrl.Finish()
+	}
+
+	if e.httpSrv != nil {
+		e.httpSrv.Close()
+	}
+
+	if e.execClient != nil {
+		// Always returns nil error
+		_ = e.execClient.Close()
+	}
+}
+
+func (e *TestEnv) setup(
+	t *testing.T,
+	ctx context.Context,
+	testAddresses []*ethcommon.Address,
+	validatorsCount uint64,
+	operatorsCount uint64,
+) error {
+	if e.followDistance == nil {
+		e.SetDefaultFollowDistance()
+	}
+	logger := zaptest.NewLogger(t)
+
+	// Create operators RSA keys
+	ops, err := createOperators(operatorsCount, 0)
+	if err != nil {
+		return err
+	}
+
+	validators := make([]*testValidatorData, validatorsCount)
+	shares := make([][]byte, validatorsCount)
+
+	// Create validators, BLS keys, shares
+	for i := 0; i < int(validatorsCount); i++ {
+		validators[i], err = createNewValidator(ops)
+		if err != nil {
+			return err
+		}
+
+		shares[i], err = generateSharesData(validators[i], ops, testAddrAlice, i)
+		if err != nil {
+			return err
+		}
+	}
+
+	eh, validatorCtrl, mockCtrl, nodeStorage, err := setupEventHandler(t, ctx, logger, ops[0], &testAddrAlice, true)
+	e.mockCtrl = mockCtrl
+	e.nodeStorage = nodeStorage
+
+	if err != nil {
+		return err
+	}
+	if validatorCtrl == nil {
+		return fmt.Errorf("error: validatorCtrl is empty")
+	}
+
+	// Adding testAddresses to the genesis block mostly to specify some balances for them
+	sim := simTestBackend(testAddresses)
+
+	// Create JSON-RPC handler
+	rpcServer, err := sim.Node.RPCHandler()
+	e.rpcServer = rpcServer
+	if err != nil {
+		return fmt.Errorf("create RPC server: %w", err)
+	}
+	// Expose handler on a test server with ws open
+	httpSrv := httptest.NewServer(rpcServer.WebsocketHandler([]string{"*"}))
+	e.httpSrv = httpSrv
+
+	addr := "ws:" + strings.TrimPrefix(httpSrv.URL, "http:")
+
+	parsed, err := abi.JSON(strings.NewReader(simcontract.SimcontractMetaData.ABI))
+	if err != nil {
+		return fmt.Errorf("parsing contract ABI: %w", err)
+	}
+
+	auth, err := bind.NewKeyedTransactorWithChainID(testKeyAlice, big.NewInt(1337))
+	if err != nil {
+		return err
+	}
+
+	contractAddr, _, _, err := bind.DeployContract(auth, parsed, ethcommon.FromHex(simcontract.SimcontractMetaData.Bin), sim)
+	if err != nil {
+		return fmt.Errorf("deploying contract: %w", err)
```

Comment:
I think initial form of verb is easier to read and as far as I know it's usually used. Let's use "deploy"